content/content.go

```diff
@@ -121,72 +165,90 @@ func parseErrorContents(ruleDirPath string) (map[string]RuleErrorKeyContent, err
 		if e.IsDir() {
 			name := e.Name()
 
-			errContent := RuleErrorKeyContent{}
-			contentFiles := []string{"generic.md"}
-			yamlFiles := []string{"metadata.yaml"}
-
-			readStrings, err := readFilesIntoString(path.Join(ruleDirPath, name), contentFiles)
-			if err != nil {
-				return errorContents, err
+			contentFiles := []string{
+				"generic.md",
+				"reason.md",
+				"metadata.yaml",
 			}
-			errContent.Generic = readStrings["generic.md"]
 
-			readBytes, err := readFilesIntoByteArray(path.Join(ruleDirPath, name), yamlFiles)
+			readContents, err := readFilesIntoFileContent(path.Join(ruleDirPath, name), contentFiles)
 			if err != nil {
 				return errorContents, err
 			}
 
-			metadataBytes := readBytes["metadata.yaml"]
-			if err := yaml.Unmarshal(metadataBytes, &errContent.Metadata); err != nil {
+			errContents, err := createErrorContents(readContents)
+			if err != nil {
 				return errorContents, err
 			}
-
-			errorContents[name] = errContent
+			errorContents[name] = *errContents
 		}
 	}
 
 	return errorContents, nil
 }
 
+// createRuleContent
+func createRuleContent(contentRead map[string]fileContent, errorKeys map[string]RuleErrorKeyContent) (*RuleContent, error) {
+	ruleContent := RuleContent{ErrorKeys: errorKeys}
+
+	if !contentRead["summary.md"].found {
+		return nil, &MissingMandatoryFile{FileName: "summary.md"}
+	}
+
+	ruleContent.Summary = string(contentRead["summary.md"].content)
+
+	if !contentRead["reason.md"].found {
```

Comment:
I think it could be refactored using a loop. But it's a nitpick and not important at all :)

Reply:
I cannot use a loop because the "reason.md" field behaves different from the rest